### ISSUE TITLE
Blocks: document current block lifecycle

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#48176]
**Issue Description:**
In the current setup, we can recognize couple of steps in the block's lifecycle:
- experimental - block is under development, sometimes functional but still behind feature flag and, available only in development build not available to users.
- beta - block is exposed to the users but with assertion that it's still under development, may be a subject to change and may occasionally malfunction
- stable - block that users can fully rely on.

We should create documentation:
- for users so they know what to expect from blocks development
- for developers so all the blocks follow the same lifecycle. Especially highlight criteria to call the block "beta" or stable etc.

**Comment by kkoteen** [comment]
Hi @kmanijak - thanks for the report.

I know this has been open for a long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've taken a look and passed this over to the Developer Advocacy team.

What I'm seeing: Latest audit evidence: Blocks/Patterns/Templates.
